src/services/factories/tokenList.ts

```diff
@@ -284,6 +288,32 @@ export function getTokensFactory(factoryParams: {
     tokenListApi.persistTokens({ networkId, tokenList: tokenDetails })
   }
 
+  async function filterAndFillTokenList(networkId: Network, tokensConfig: TokenDetails[]): Promise<TokenDetails[]> {
+    const tokensWithNoDetails: TokenDetails[] = []
+    const tokensWithDetails = tokensConfig.reduce<TokenDetails[]>((acc, token) => {
+      if (!token.name || !token.symbol) {
+        tokensWithNoDetails.push(token)
+        return acc
+      }
+
+      acc.push(token)
+      return acc
+    }, [])
```

Comment:
Not sure there's much advantage to `reduce` over a for loop if you do side effects (`tokensWithNoDetails.push(token)`) anyways